internal/source/image.go

```diff
@@ -190,3 +181,60 @@ func (i *Image) handleUnexpectedPod(ctx context.Context, pod *corev1.Pod) error
 	_ = i.Client.Delete(ctx, pod)
 	return fmt.Errorf("unexpected pod phase: %v", pod.Status.Phase)
 }
+
+func pendingImagePodResult(pod *corev1.Pod) *Result {
+	var messages []string
+	for _, cStatus := range append(pod.Status.InitContainerStatuses, pod.Status.ContainerStatuses...) {
+		if waiting := cStatus.State.Waiting; waiting != nil {
+			if waiting.Reason == "ErrImagePull" || waiting.Reason == "ImagePullBackOff" {
+				messages = append(messages, waiting.Message)
+			}
+		}
+	}
+	return &Result{State: StatePending, Message: strings.Join(messages, "; ")}
+}
+
+// addSecurityContext is responsible for taking a container and defining the
+// relevant security context values. By having a function do this, we can keep
+// that configuration easily consistent and maintainable.
+func addSecurityContext(pod *corev1.Pod) {
```

Comment:
Do we need to nil check pod or pod.Spec here?

Reply:
I don't think it would hurt, there could be a situation where this function gets broken out into another package and is passed a nil pod. Added in the latest commit.